ledger-core-bitcoin/src/bitcoin/keychains/CommonBitcoinLikeKeychains.cpp

```diff
@@ -227,6 +227,18 @@ namespace ledger {
             return getAddressDerivationPath(address).nonEmpty();
         }
 
+        std::vector<BitcoinLikeKeychain::Address> CommonBitcoinLikeKeychains::getAllAddresses() {
+            std::vector<BitcoinLikeKeychain::Address> result;
+            result.reserve(_state.maxConsecutiveChangeIndex + 1 + _state.maxConsecutiveReceiveIndex + 1);
+            for (auto i = 0; i <= _state.maxConsecutiveChangeIndex; i++) {
+                result.push_back(derive(KeyPurpose::CHANGE, i));
+            }
+            for (auto i = 0; i <= _state.maxConsecutiveReceiveIndex; i++) {
+                result.push_back(derive(KeyPurpose::RECEIVE, i));
+            }
+            return result;
```

Comment:
You don't need the two extra slots when reserving memory. BTW I suggest you to remove redundant codes even in local function for readability, maintenance and to ease the cognitive complexity. I also suggest to use better name for variables - `result` name is too generic for such a case.

```suggestion
            std::vector<BitcoinLikeKeychain::Address> addresses;
            addresses.reserve(_state.maxConsecutiveChangeIndex + _state.maxConsecutiveReceiveIndex);
            
            auto fetchAddressesFrom = [&addresses](auto const keyPurpose, auto const maxIndex) {
                  for (auto i = 0; i <= maxIndex; ++i) {
                      addresses.push_back(derive(keyPurpose, i));
                  }
            };
            
            fetchAddressesFrom(KeyPurpose::CHANGE, _state.maxConsecutiveChangeIndex);
            fetchAddressesFrom(KeyPurpose::RECEIVE, _state.maxConsecutiveReceiveIndex);
            
            return addresses;
```

Reply:
- for (auto i = 0; i <= maxIndex; ++i) => maxIndex+1 items
- Do we really need a delta expression just for a loop ? For me, it affects fluidity when reading the code